### ISSUE TITLE
Error running bosh commands after a bbl up

### ISSUE THREAD

**Issue by dawu415** [cloudfoundry/bosh-bootloader#304]
Hi,

This worked for me a few weeks ago.  This is for AWS. I am running on OSX Sierra (10.12.6)

I did 

```
bbl plan --iaas aws --aws-access-key-id <xxxx> --aws-secret-access-key <zzz> --aws-region us-east-1 --lb-type concourse
```

and then

```
bbl up --iaas aws --aws-access-key-id <xxxx> --aws-secret-access-key <zzz> --aws-region us-east-1 --lb-type concourse
```

then I could run `bosh vms`  etc.   Cleaning up: I ran a bbl destroy command, which completed successfully.  

However, when I tried to run the bbl steps above again (in a new directory). It deploys without any errors. I then try to run `bosh vms` (or any valid bosh commands) I get the following error

```
Validating Director connection config:
  Parsing certificate 1: Missing PEM block

Exit code 1
```

I was on bbl 6.6.4.  I tried updating to 6.6.7 but still no luck. 

Was there something new or some extra steps I have to take?

**Comment by genevieve** [comment]
Is `BOSH_CA_CERT` set in your environment? Did you run `eval "$(bbl print-env)"`?

**Comment by dawu415** [comment]
Yes, I ran `eval "$(bbl print-env)"` first and also checked BOSH_CA_CERT thereafter. The environment variable is there. 

Some debugging steps I took:
1) Start new terminal, run `export` to ensure there are no BBL environment variables
2) ran the `bbl up` command with parameters above
3) After completion, run `export` to ensure that no BBL environment variables are there
4) Execute `eval "$(bbl print-env)"`
5) Run `export` and can confirm that all the BBL environment variables are there, including `BOSH_CA_CERT`
6) Confirm that the BOSH_CA_CERT matches that in the director vars store yml, under /default_ca/ca 

```
BOSH_ALL_PROXY='ssh+socks5://jumpbox@<el-ipaddy>:22?private-key=/var/folders/45/n3wsmxkj61n4vssl53q3lr7h0000gn/T/bosh-zzz/bosh_jumpbox_private.key'
BOSH_CA_CERT=<bosh ca>
BOSH_CLIENT=admin
BOSH_CLIENT_SECRET=<secret>
BOSH_ENVIRONMENT=https://<boshdir-ipaddy>:25555
CREDHUB_CA_CERT=<credhub_certs>
CREDHUB_CLIENT=credhub-admin
CREDHUB_PROXY='ssh+socks5://jumpbox@<el-ipaddy>:22?private-key=/var/folders/45/zzz/T/bosh-zzzz/bosh_jumpbox_private.key'
CREDHUB_SECRET=zzzz
CREDHUB_SERVER=https://<boshdir-ipaddy>:8844
```

**Comment by actdiux** [comment]
I have the exact same issue on Azure. I am a complete n00b and initially just thought I was doing something wrong.

**Comment by genevieve** [comment]
@actdiux Could you try 
```
export BBL_STATE_DIR=$ARA_STATE_BBL
eval "$(bbl print-env)"
```

**Comment by actdiux** [comment]
# 20 years a Linux admin and the quotes gets me!

forever grateful genevieve!

**Comment by scottbri** [comment]
https://github.com/cloudfoundry/bosh-bootloader/blob/master/docs/howto-target-bosh-director.md
Every time I reload this page, the quotes are always there.  I just didn't see.  Perhaps a bit of text to highlight a "common gotcha" of forgetting the quotes, esp. since the alternatives to bbl print-env don't use the quotes?

**Comment by astrieanna** [comment]
It would be great to stick this tip in the CLI output with this error.

**Comment by nmahoney-pivotal** [comment]
I just called out in the docs that the quotes are necessary: https://github.com/cloudfoundry/bosh-bootloader/blob/master/docs/howto-target-bosh-director.md

@astrieanna which error output are you talking about? It's a little tricky because there is no `bbl` error, and to `bosh` it just looks like the `BOSH_CA_CERT` is bad. `bosh` could maybe say something when `BOSH_CA_CERT` is bad like `Did you target this director with 'eval "$(bbl print-env)"'? Those quotes are necessary!` I'm not sure if they would want to add something like that.

**Comment by astrieanna** [comment]
That's a good point. I did mean the BOSH error; I can see the bosh CLI maybe not wanting to give a tip that might be totally off-base if you didn't use BBL. It's probably not possible to really know how many people encounter that bosh error for reasons other than forgetting quotes on their bbl command.

**Comment by astrieanna** [comment]
That's a good point. I did mean the BOSH error; I can see the bosh CLI maybe not wanting to give a tip that might be totally off-base if you didn't use BBL. It's probably not possible to really know how many people encounter that bosh error for reasons other than forgetting quotes on their bbl command.

**Comment by astrieanna** [comment]
You might want to add the literal error message
```
Validating Director connection config:
  Parsing certificate 1: Missing PEM block

Exit code 1
```
to your documentation. The first thing I did with that error was google it, and that lead me to this issue. If you include that text in your docs (either near the command that needs quotes or in an FAQ/troubleshooting section), then I would be more likely to find the docs rather than this issue when debugging.

**Comment by pkammari** [comment]
i have landed on same error and i am running bosh in WSL and i cant install bbl, any help is much appreciated. 

  `root@LT-9888K13:~/ca-central1# ./bosh -v
version 6.4.8-cabff008-2021-11-01T16:53:16Z

Succeeded`

`root@LT-9888K13:~/ca-central1# ./bosh env
Validating Director connection config:
  Parsing certificate 1: Missing PEM block

Exit code 1`

**Comment by pkammari** [comment]
> i have landed on same error and i am running bosh on WSL and i cant install bbl, any help is much appreciated.
> 
> `root@LT-9888K13:~/ca-# ./bosh -v version 6.4.8-cabff008-2021-11-01T16:53:16Z
> 
> Succeeded`
> 
> `root@LT-9888K13:~/ca-# ./bosh env Validating Director connection config: Parsing certificate 1: Missing PEM block
> 
> Exit code 1`

this issue may cause in WSL due to wrong indentation or spaces or ^M characters in the cert file.
